docs/docs/graphql/fragments.md

```diff
@@ -310,3 +310,66 @@ To generate the `src/graphql/possibleTypes` file, enable fragments in `redwood.t
 [graphql]
   fragments = true
 ```
+
+## Testing, Storybook and Mock Data
+
+When using fragments with test or Storybook, it is important that you define your mock data correctly.
+
+By including the `__typename` and the GraphQL Type for the mocked data object, you ensure that Apollo Client will properly handle the information if using the cache or fragments.
+
+For example, consider the fragment `BookInfo` used by the query `GetBookDetails`.
+
+```ts
+import type { Book } from 'types/graphql'
+
+import { registerFragment } from '@redwoodjs/web/apollo'
+
+const { useRegisteredFragment } = registerFragment(
+  gql`
+    fragment BookInfo on Book {
+      id
+      title
+      author
+      publicationYear
+    }
+  `
+)
+```
+
+```ts
+import type { GetBookDetails } from 'types/graphql'
+
+import { useQuery } from '@redwoodjs/web'
+
+import BookInfo from 'src/components/BookInfo'
+
+const GET_BOOK_DETAILS = gql`
+  query GetBookDetails($bookId: ID!) {
+    book(id: $bookId) {
+      ...BookInfo
+      description
+      # Include other fields specific to this query
+    }
+  }
+```
+
+### Mock Data
+
+To satisfy the query as well as the fragment needs your mock data should include the `Book` typename:
```

Comment:
Don't know if this just needs a comma somewhere but I can't wrap my head around this sentence sorry.